app/javascript/packages/request/index.spec.tsx

```diff
@@ -0,0 +1,124 @@
+import { useSandbox } from '@18f/identity-test-helpers';
+import { request } from '.';
+import '@18f/identity-polyfill';
+
+describe('request', () => {
+  const sandbox = useSandbox();
+
+  it('includes the CSRF token by default', async () => {
+    const csrf = 'TYsqyyQ66Y';
+    const mockGetCSRF = () => csrf;
+
+    sandbox.stub(window, 'fetch').callsFake((url, init = {}) => {
+      const headers = init.headers as Headers;
+      expect(headers.get('X-CSRF-Token')).to.equal(csrf);
+
+      return Promise.resolve(
+        new Response(new Blob([JSON.stringify({})]), {
+          status: 200,
+        }),
+      );
+    });
+
+    await request('https://example.com', {
+      csrf: mockGetCSRF,
+    });
```

Comment:
This test doesn't guarantee that `window.fetch` was actually called, and since that's where we run the assertions, this may not give us a ton of confidence if it happens that `window.fetch` was never called.

I'd suggest either adding an assertion of the callCount:

```suggestion
    await request('https://example.com', {
      csrf: mockGetCSRF,
    });

    expect(window.fetch).to.have.been.calledOnce();
```

Or we could reform this as a [sinon mock](https://sinonjs.org/releases/latest/mocks/) and assert an expectation of it being called (to be fair, we don't use this Sinon feature very often)